src/components/scenes/GiftCardMarketScene.tsx

```diff
@@ -0,0 +1,640 @@
+import { debugLog } from '../../util/logger'
+import { CountryButton } from '../buttons/RegionButton'
+import { EdgeCard } from '../cards/EdgeCard'
+import { GiftCardTile } from '../cards/GiftCardTile'
+import { CircularBrandIcon } from '../common/CircularBrandIcon'
+import { EdgeAnim } from '../common/EdgeAnim'
+import { EdgeTouchableOpacity } from '../common/EdgeTouchableOpacity'
+import { SceneWrapper } from '../common/SceneWrapper'
+import { GridIcon, ListIcon } from '../icons/ThemedIcons'
+import { SceneContainer } from '../layout/SceneContainer'
+import { normalizeCategory } from '../modals/GiftCardSearchModal'
+import { FillLoader } from '../progress-indicators/FillLoader'
+import { showError } from '../services/AirshipInstance'
+import { cacheStyles, type Theme, useTheme } from '../services/ThemeContext'
+import { EdgeText } from '../themed/EdgeText'
+import { SearchFooter } from '../themed/SearchFooter'
+
+type ViewMode = 'grid' | 'list'
+
+// Internal constant for "All" category comparison - display uses lstrings.string_all
+const CATEGORY_ALL = 'All'
+
+/**
+ * Formats a normalized category for display:
+ * - Replaces dashes with " & "
+ * - Capitalizes first letter of each word
+ */
+const formatCategoryDisplay = (category: string): string => {
+  return category
+    .split('-')
+    .map(word => word.charAt(0).toUpperCase() + word.slice(1))
+    .join(' & ')
+}
+
+interface MarketItem {
+  brandName: string
+  priceRange: string
+  productId: number
+  productImage: string
+  categories: string[]
+  isBitrefill?: boolean
+}
+
+/**
+ * Formats a price range string from brand data.
+ * - Variable range: shows "min - max" format
+ * - Fixed denominations: shows comma-separated list
+ */
+const formatPriceRange = (brand: PhazeGiftCardBrand): string => {
+  const { currency, valueRestrictions, denominations } = brand
+  const { minVal, maxVal } = valueRestrictions
+
+  // Variable range - show min to max
+  if (minVal != null && maxVal != null) {
+    return `${minVal} - ${maxVal} ${currency}`
+  }
+
+  // Fixed denominations - show comma-separated list
+  if (denominations.length > 0) {
+    const sorted = [...denominations].sort((a, b) => a - b)
+    return `${sorted.join(', ')} ${currency}`
+  }
+
+  return currency
+}
+
+// Bitrefill partner option shown at end of results
+const BITREFILL_ITEM: MarketItem = {
+  brandName: 'Bitrefill',
+  priceRange: lstrings.gift_card_more_options,
+  productId: -999,
+  productImage: `${EDGE_CONTENT_SERVER_URI}/bitrefill.png`,
+  categories: [],
+  isBitrefill: true
+}
+
+interface Props extends EdgeAppSceneProps<'giftCardMarket'> {}
+
+export const GiftCardMarketScene: React.FC<Props> = (props: Props) => {
+  const { navigation } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
+  const dispatch = useDispatch()
+
+  // Get user's current country settings
+  const { countryCode } = useSelector(state => state.ui.settings)
```

Comment:
Do `const countryCode = useSelector(state => state.ui.settings.countryCode)`. Otherwise *any* setting change will render the scene.